test/jdk/java/lang/System/i18nEnvArg.java

```diff
@@ -0,0 +1,155 @@
+        }
+    }
+
+    public static class Start {
+        public static void main(String[] args) throws Exception {
+            String nativeEncoding = System.getProperty("native.encoding");
+            Charset dcs = nativeEncoding == null ?
+                Charset.defaultCharset() :
+                Charset.forName(nativeEncoding);
+            Charset cs = Charset.forName("x-euc-jp-linux");
+            if (!dcs.equals(cs)) {
+                return;
+            }
+            ProcessBuilder pb = new ProcessBuilder(javeExe,
+                "--add-opens=java.base/java.lang=ALL-UNNAMED",
+                "-classpath",
+                System.getProperty("java.class.path"),
+                "i18nEnvArg$Verify",
+                text);
+            pb.redirectErrorStream(true);
+            Map<String, String> environ = pb.environment();
+            environ.clear();
+            environ.put("LANG", "ja_JP.eucjp");
+            environ.put(text, text);
+            Process p = pb.start();
+            InputStream is = p.getInputStream();
+            byte[] ba = new byte[maxSize];
+            ByteBuffer bb = ByteBuffer.wrap(ba);
+            int ch;
+            while((ch = is.read()) != -1) {
+                bb.put((byte)ch);
+            }
+            int rc = p.waitFor();
+            System.out.write(ba, 0, bb.position());
+        }
+    }
+
+    public static class Verify {
+        public static void main(String[] args) throws Exception {
+            Charset cs = Charset.forName("x-euc-jp-linux");
+            byte[] kanji = (text + "=" + text).getBytes(cs);
+            String s = System.getenv(text);
+            if (!text.equals(s)) {
+                throw new Exception("getenv() returns unexpected data");
+            }
+            if (!text.equals(args[0])) {
+                System.out.print("Unexpected argument was received: ");
+                for(char ch : text.toCharArray()) {
+                   System.out.printf("\\u%04X", (int)ch);
+                }
+                System.out.print("<->");
+                for(char ch : args[0].toCharArray()) {
+                   System.out.printf("\\u%04X", (int)ch);
+                }
+                System.out.println();
+            }
+            Class<?> cls = Class.forName("java.lang.ProcessEnvironment");
+            Method enviorn_mid = cls.getDeclaredMethod("environ");
+            enviorn_mid.setAccessible(true);
+            byte[][] environ = (byte[][]) enviorn_mid.invoke(null,
+                (Object[])null);
+            byte[] ba = new byte[maxSize];
+            StringJoiner sj = new StringJoiner(", ");
+            for(int i = 0; i < environ.length; i += 2) {
+                ByteBuffer bb = ByteBuffer.wrap(ba);
+                bb.put(environ[i]);
+                bb.put((byte)'=');
+                bb.put(environ[i+1]);
+                byte[] envb = Arrays.copyOf(ba, bb.position());
+                if (Arrays.equals(kanji, envb)) return;
```

Comment:
It seems odd to exit the loop on the first match.

I think AIX always has environment variables not set by the caller.

Reply:
On this testing, use 2 environment variables:
* LANG=ja_JP.eucjp
* \u6F22\u5B57=\u6F22\u5B57

If the other environment variable is defined, it's printed.